### PR TITLE
Fix 500 error when genomicDataFilters is empty (#11206)

### LINKED ISSUE #11206: 500 error evaluating genomicDataFilter
Resulting in a 500 error
```
curl 'http://localhost:8082/api/column-store/filtered-samples/fetch' \
  -H 'Accept: application/json' \
  -H 'Accept-Language: en-US,en;q=0.9' \
  -H 'Cache-Control: no-cache' \
  -H 'Connection: keep-alive' \
  -H 'Content-Type: application/json' \
  -H 'Cookie: _ga=GA1.1.1887007066.1710956751; _ga_5260NDGD6Z=GS1.1.1721930557.11.1.1721930574.0.0.0' \
  -H 'Origin: http://localhost:8082' \
  -H 'Pragma: no-cache' \
  -H 'Referer: http://localhost:8082/study/summary?id=nsclc_public_genie_bpc' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36' \
  -H 'sec-ch-ua: "Chromium";v="130", "Google Chrome";v="130", "Not?A_Brand";v="99"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  --data-raw '{"genomicDataFilters":[{"hugoGeneSymbol":"TP53","profileType":"cna"}],"studyIds":["nsclc_public_genie_bpc"],"alterationFilter":{"copyNumberAlterationEventTypes":{"AMP":true,"HOMDEL":true},"mutationEventTypes":{"any":true},"structuralVariants":null,"includeDriver":true,"includeVUS":true,"includeUnknownOncogenicity":true,"includeUnknownTier":true,"includeGermline":true,"includeSomatic":true,"includeUnknownStatus":true,"tiersBooleanMap":{}}}'
```

### PR DESCRIPTION
**Pull request by heathdutton**
Fix #11206

Describe changes proposed in this pull request:
- Add null/empty validation for `genomicDataFilters` in `fetchGenomicDataCounts` and `fetchMutationDataCounts` endpoints
- Return empty list instead of propagating IndexOutOfBoundsException from `MyBatis` mapper
- Add unit tests covering empty and null filter scenarios

# Checks
- [x] The commit log is comprehensible. It follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). We can fix this during merge by using a squash+merge if necessary
- [x] Has tests or has a separate issue that describes the types of test that should be created. If no test is included it should explicitly be mentioned in the PR why there is no test.
- [ ] Is this PR adding logic based on one or more **clinical** attributes? If yes, please make sure validation for this attribute is also present in the data validation / data loading layers (in backend repo) and documented in [File-Formats Clinical data section](https://github.com/cBioPortal/cbioportal/blob/master/docs/File-Formats.md#clinical-data)!
- [ ] Make sure your PR has one of the labels defined in https://github.com/cBioPortal/cbioportal/blob/master/.github/release-drafter.yml

# Any screenshots or GIFs?
NA

# Notify
@alisman @inodb

### CODE REVIEW

File: src/main/java/org/cbioportal/application/rest/vcolumnstore/ColumnarStoreStudyViewController.java [hunk 1]
```diff
@@ -424,6 +425,13 @@ public ResponseEntity<List<GenomicDataCountItem>> fetchGenomicDataCounts(
       throws StudyNotFoundException {
     List<GenomicDataFilter> genomicDataFilters = genomicDataCountFilter.getGenomicDataFilters();
     StudyViewFilter studyViewFilter = genomicDataCountFilter.getStudyViewFilter();
+
+    // If genomicDataFilters is null or empty, return an empty result list
+    // This prevents IndexOutOfBoundsException in the MyBatis mapper
+    if (genomicDataFilters == null || genomicDataFilters.isEmpty()) {
+      return new ResponseEntity<>(Collections.emptyList(), HttpStatus.OK);
```
Comment (onursumer) [inline src/main/java/org/cbioportal/application/rest/vcolumnstore/ColumnarStoreStudyViewController.java]: Might be better to use the `ok` method here as well for consistency.

```suggestion
      return ResponseEntity.ok(Collections.emptyList());
```

File: src/test/java/org/cbioportal/application/rest/vcolumnstore/ColumnarStoreStudyViewControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    private ObjectMapper objectMapper;
+
+    @MockBean
+    private StudyViewService studyViewService;
+
+    private static final String TEST_STUDY_ID = "test_study";
+
+    /**
+     * Test that fetchGenomicDataCounts returns empty list when genomicDataFilters is empty.
+     * This verifies the fix for Issue #11206.
```
Comment (onursumer) [inline src/test/java/org/cbioportal/application/rest/vcolumnstore/ColumnarStoreStudyViewControllerTest.java]: we probably don't need this line of comment with issue reference. same applies to the other tests below.

**Review by alisman — APPROVED** [review]

**Comment by onursumer** [comment]
@copilot can you fix the spotless issues?

**Comment by onursumer** [comment]
The newly added `ColumnarStoreStudyViewControllerTest` is failing. See https://github.com/cBioPortal/cbioportal/actions/runs/23764323913/job/69239842565?pr=11876

**Comment by alisman** [comment]
@heathdutton thanks for the fix.   there are now some sonar issues to look at:

https://sonarcloud.io/project/issues?id=cBioPortal_cbioportal&pullRequest=11876&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true

**Comment by onursumer** [comment]
There are conflicts. We will reimplement with an agent.
